crates/dslab-mp/src/mc/node.rs

```diff
@@ -113,7 +116,11 @@ impl McNode {
         proc_entry.received_message_count += 1;
 
         let mut proc_ctx = Context::basic(proc.to_string(), time, self.clock_skew, random_seed);
-        proc_entry.proc_impl.on_message(msg, from, &mut proc_ctx);
+
+        if let Err(e) = proc_entry.proc_impl.on_message(msg, from, &mut proc_ctx) {
+            self.handle_process_err(e, proc.clone());
+        }
```

Comment:
Мне больше нравится такой вариант, когда в начале идет вызов процесса, а далее обрабатывается ошибка:

```suggestion
        proc_entry
            .proc_impl
            .on_message(msg, from, &mut proc_ctx)
            .map_err(|e| self.handle_process_err(e, proc.clone()))
            .unwrap();
```

Reply:
Только unwrap вроде лишний